main/social/personal_data.php

```diff
@@ -30,8 +30,9 @@
 
 $action = isset($_REQUEST['action']) ? $_REQUEST['action'] : '';
 $formToString = '';
-
-if (api_get_setting('allow_terms_conditions') === 'true') {
+$w = api_get_setting('allow_terms_conditions');
+$w = 'true';
+if ($w === 'true') {
```

Comment:
$w will always be equal to true because you just set it just before. There is something wrong here.
Also I do not see the relation between this change and the aim of this commit.